backend/qemu.pm

```diff
@@ -418,29 +420,33 @@ sub start_qemu() {
             }
         }
 
+        if ($vars->{CDMODEL} =~ /virtio-scsi.*/) {
+            # add scsi controller if hdd didn't do it already
+            if ($vars->{CDMODEL} ne $vars->{HDDMODEL}) {
+                push(@params, "-device", "$vars->{CDMODEL},id=scsi0");
+            }
+            $vars->{CDMODEL} = "scsi-cd";
+        }
+
+
         if ($iso) {
             if ($vars->{USBBOOT}) {
                 push(@params, "-drive",  "if=none,id=usbstick,file=$iso,snapshot=on");
                 push(@params, "-device", "usb-ehci,id=ehci");
                 push(@params, "-device", "usb-storage,bus=ehci.0,drive=usbstick,id=devusb");
             }
-            elsif ($vars->{CDMODEL}) {
+            else {
                 push(@params, '-drive',  "media=cdrom,if=none,id=cd0,format=raw,file=$iso");
```

Comment:
indeed. meanwhile I found that adding snapshot=on here breaks loading cd drives again which is needed for zdup on uefi o_O.